app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -455,60 +489,74 @@ public function getQuoteById($quoteId)
      */
     protected function validateCoupons(Mage_Sales_Model_Quote $immutableQuote, $transaction) {
 
-        if (!@$transaction->order->cart->discounts) {
-            return;
-        }
-
-        /*
-         * Natively, Magento only supports one coupon code per order, but we can build
-         * basic support here for plugins like Amasty or custom solutions that implement
-         * multiple coupons.
-         *
-         * Here, we use "," to delimit multiple coupon codes like Amasty and popular multi-coupon
-         * custom code strategies.  This implementation also supports standard magento single coupon
-         * format.
-         */
-        foreach($transaction->order->cart->discounts as $boltCoupon) {
-
-            if (@$boltCoupon->reference) {
-                $magentoCoupon = Mage::getModel('salesrule/coupon')->load($boltCoupon->reference, 'code');
-                $couponExists = (bool) $magentoCoupon->getId();
-
-                if ($couponExists) {
-
-                    $magentoCouponCodes = $immutableQuote->getCouponCode() ? explode(',', (string) $immutableQuote->getCouponCode()) : array();
+        Mage::dispatchEvent(
```

Comment:
why not simply return if `!@$transaction->order->cart->discounts`? do we still need to call these events?